src/layout/layoutfmt.rs

```diff
@@ -17,20 +16,18 @@ use std::fmt;
 impl fmt::Debug for Layout {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         if self.0 == 0 {
-            write!(f, "Custom")
+            write!(f, "Custom")?
         } else {
-            write!(
-                f,
-                "{}",
-                (0..32)
-                    .filter(|&i| self.is(1 << i))
-                    .format_with(" | ", |i, f| if let Some(name) = LAYOUT_NAMES.get(i) {
-                        f(name)
+            (0..32)
+                .filter(|&i| self.is(1 << i))
+                .try_fold((), |_, i| {
+                    if let Some(name) = LAYOUT_NAMES.get(i) {
+                        write!(f, "{}", name)
                     } else {
-                        f(&format_args!("0x{:x}", i))
-                    })
-            )
-        }?;
+                        write!(f, "{:#x}", i)
+                    }
+                })?;
```

Comment:
With `.try_fold()` the `Itertools::format` method is no longer as indispensible as it was.